src/payment/strategies/braintree/braintree-credit-card-payment-strategy.ts

```diff
@@ -177,6 +191,35 @@ export default class BraintreeCreditCardPaymentStrategy implements PaymentStrate
         };
     }
 
+    private async _processAdditionalAction(
+        error: Error,
+        payment: OrderPaymentRequestBody,
+        billingAddress: Address,
+        orderAmount: number
+    ): Promise<InternalCheckoutSelectors> {
+        // test: check handlerError function
+        if (!(error instanceof RequestError) || !some(error.body.errors, { code: 'three_d_secure_required' })) {
+            return this._handleError(error);
+        }
+
+        try {
+            const { payer_auth_request: storedCreditCardNonce } = error.body.three_ds_result || {};
+
+            return await this._store.dispatch(this._paymentActionCreator.submitPayment({
+                ...payment,
+                paymentData: {
+                    deviceSessionId: this._deviceSessionId,
+                    ...this._mapToNonceInstrument({
+                        ...payment.paymentData,
+                        ...await this._braintreePaymentProcessor.challenge3DSVerification(storedCreditCardNonce, orderAmount),
+                    }),
+                },
+            }));
+        } catch (error) {
+            return this._processAdditionalAction(error, payment, billingAddress, orderAmount);
```

Comment:
This seems like some kind of recursion, where if the submitPayment fails then we call the method again. Does that seem right? Since it can run into an infinite loop.

Reply:
It can run into an infinite loop only in case if the backend will return the same "3DS required" error. This method are used to catch another errors that can be thrown in case of backend or braintree issues when we try to submit order with verified data.

In [this case](https://github.com/bigcommerce/checkout-sdk-js/pull/1410/files/b77e28e2eb93a851505b95c6b8935fb9b27d3680#diff-481762847de0d12fd9cbf38ac9e0555aca512bbdda28be8437d3320bfbdcca14R201) we will throw an error as a modal to the customer through checkout-js, otherwise we should trigger 3DS verification for stored credit card.

Reply:
Perhaps, I can change it to handleError function to prevent infinite loop

Reply:
Fixed